### ISSUE TITLE
Support for Pipelining

### ISSUE THREAD

**Issue by KodrAus** [tailhook/rotor-http#47]
Just a quick question: is http pipelining supported on persistent client connections in `rotor_http`?

I'm not an expert on this particular feature, just starting to look into it, so I'm not sure if the complete set of requests to pipeline need to be sent upfront or whether they can be thrown onto the request stream at any time.

**Comment by pyfisch** [comment]
Yes it is supported. You can send all requests upfront or wait varying amounts of time between each.

**Comment by KodrAus** [comment]
Ah nice, so am I just able to return `Task::Request` from my state machine multiple times to pipeline requests or is there something else I need to do?
